src/megaapi_impl.cpp

```diff
@@ -8726,20 +8721,7 @@ void MegaApiImpl::sendPendingRequests()
                     attrvalue = value;
                     break;
                 }
-
-                case MegaApi::USER_ATTR_AUTHRING:
-                {
-                    attrname = "*!authring";
-                    attrvalue = value;
-                    break;
-                }
-
-                case MegaApi::USER_ATTR_LAST_INTERACTION:
-                {
-                    attrname = "*!lstint";
-                    attrvalue = value;
-                    break;
-                }
+                // authentication ring and last interaction are ro attributes
```

Comment:
Currently, no SDK client is using the last interaction value or the authentication ring (yet). I suggest to keep those changes for a future branch and to release this code to generate the required keys for chat without lstint and authring.
In any case, the missing part in the SDK is the ability to set those attributes, but not to read them. Every private attribute is TLV-decoded after decryption (`CommandGetUA::procresult()`)